src/box/txn.c

```diff
@@ -885,6 +896,19 @@ txn_journal_entry_new(struct txn *txn)
 
 		req->approx_len += xrow_approx_len(stmt->row);
 	}
+
+	/*
+	 * Look at the flags stmt->row->flags. If the transaction
+	 * is synchronous, then set is_sync = true. This should
+	 * only be done on replicas. The master sets these flags
+	 * and independently decides whether the transaction is
+	 * synchronous or not. All txn meta flags are set only for
+	 * the last txn row.
+	 */
+	stmt = stailq_last_entry(&txn->stmts, typeof(*stmt), next);
+	if ((stmt->row->flags & IPROTO_FLAG_WAIT_ACK) != 0)
+		is_sync = true;
+
```

Comment:
This number of new `if` and a new `bool` flag in the txn object don't look good at all given how hot the code is. Why couldn't you set `TXN_WAIT_ACK` flag in `make_sync()`? It doesn't seem that you need a new boolean member in `struct txn`. It is only confusing now, that we have the "sync"-property spread across 2 members (flags and the new bool).